src/NetSapiensSharp/Connector.cs

```diff
@@ -121,18 +124,30 @@ public IRestResponse<Tresponse> Send<Tresponse>(Request<Tresponse> request)
                     break;
                 }
             }
-            while (_SessionToken == null && myRetryCount < _UnauthorizedRetryLimit);
+            while (_SessionToken == null && ((myAuthorizationRetryCount > 0 && myAuthorizationRetryCount < _UnauthorizedRetryLimit) || (myErrorRetryCount > 0 && myErrorRetryCount < _ErroredRetryLimit)));
 
-            if (myRetryCount == _UnauthorizedRetryLimit)
+            if (myAuthorizationRetryCount == _UnauthorizedRetryLimit)
             {
                 myResponse.Data = null;
                 myResponse.ErrorException = null;
                 myResponse.ErrorMessage = $"Failed to authorize. Retry limit of {_UnauthorizedRetryLimit} has been reached.";
                 myResponse.ResponseStatus = ResponseStatus.Aborted;
             }
+            else if (myErrorRetryCount == _ErroredRetryLimit)
+            {
+                myResponse.Data = null;
+                myResponse.ErrorException = null;
+                myResponse.ErrorMessage = $"Failed to receive a successful response. Retry limit of {_ErroredRetryLimit} has been reached.";
+                myResponse.ResponseStatus = ResponseStatus.Error;
+            }
             return myResponse;
         }
 
+        private int CalculateWait(int myIteration)
+        {
+            return (int)(Math.Round(0.21412 + 0.155521 * Math.Exp(1.05038 * myIteration), 3) * 60000);
```

Comment:
This looks like a very specific algorithm we pulled from somewhere, but there is nothing to indicate what any of these values mean.

Can you provide a reference of what this is doing?

Reply:
I got the inspiration from Defines on this. Essentially, we are exponentially increasing the wait time as the retry limit increases. The idea is that we would keep retrying and wait longer after each time if we were not able to successfully connect.
Here is some sample code I wrote up that lists the times:
```
for (int i = 1; i <= 5; i++) 
  {
    int delayInMilliseconds = (int)(Math.Round(0.21412 + 0.155521 * Math.Exp(1.05038 * i), 3) * 60000);
    var delayInSeconds = Math.Round(delayInMilliseconds/1000.00,2);
    var delayInMinutes = Math.Round(delayInSeconds/60.00,2);
    Console.WriteLine("Time to wait for attempt " + i.ToString() + ": "+ delayInSeconds.ToString() + " seconds or " + delayInMinutes + " minutes.");
  }
```
Here are the results:
![image](https://user-images.githubusercontent.com/93213727/148713199-1d784154-e8ca-4011-8cb7-68c4b38dc275.png)

Reply:
Sorry for any confusion - I understand that this is [exponential backoff](https://en.wikipedia.org/wiki/Exponential_backoff) (a pretty common solution for this kind of thing). That said, those backoff numbers are _very aggressive_ - waiting 39 seconds to start on iteration one only to wait nearly 30 minutes for time two is beyond excessive - that would be an experience that would have all kinds of issues.

Above that, what I'm really struggling with is _where you came up with these numbers_ and how someone might be able to interpret this later. [Magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)) are something to ultimately avoid, so we'd want to document & otherwise name the components of this if at all possible.

I was also thinking that Defines might have talked about this with you, but I would think he talked about [Polly (a Robust .NET retry library)](https://github.com/App-vNext/Polly) and not rolling our own solution for exponential backoff.